Tasks/NuGetInstaller/nugetinstaller.ts

```diff
@@ -65,8 +66,9 @@ if (!tl.filePathSupplied('nuGetPath')) {
 var serviceUri = tl.getEndpointUrl("SYSTEMVSSCONNECTION", false);
 
 //find nuget location to use
-var nuGetPathToUse = ngToolRunner.locateTool('nuget.exe', userNuGetPath);
-var credProviderPath = ngToolRunner.locateTool('CredentialProvider.TeamBuild.exe', null, true);
+var nuGetExeVariants = ['nuget.exe', 'NuGet.exe', 'nuget', 'NuGet']
+var nuGetPathToUse = ngToolRunner.locateTool(nuGetExeVariants, {userPath: userNuGetPath, fallbackToSystemPath: os.platform() !== 'win32'});
```

Comment:
Have you considered if the user specifies the "Path to NuGet.exe" as the path to the file name? It looks like locateTool expects the user path to be a directory path. The verbiage on the input seems to indicate a file path.